src/pk.c

```diff
@@ -10316,7 +10316,7 @@ size_t wolfSSL_EC_POINT_point2oct(const WOLFSSL_EC_GROUP *group,
             /* Check return. When buf is NULL, return will be length only
              * error.
              */
-            if (ret != ((buf != NULL) ? MP_OKAY : LENGTH_ONLY_E)) {
+            if (ret != ((buf != NULL) ? (int)MP_OKAY : (int)LENGTH_ONLY_E)) {
```

Comment:
This is getting ridiculous, why cast constants to ints!
A negative at that with LENGTH_ONLY_E!
What is it complaining about?

Reply:
This is incorrect code in `master` that is mixing two namespaces, the `error-crypt.h` namespace and the MP error code namespace.  I didn't want to embark on a wider-ranging effort to fix these, so I taped it over with the `int` casts.  `g++` by default will correctly warn on the ternary operator choosing between clashing `enum`s.  it didn't warn before because the MP codes weren't `enum`s before.  making the MP codes `enum`s allowed the compiler to correctly detect the namespace violation.

I discern two correct ways forward, (1) the two namespaces can be merged together, since currently there is no conflict between them, or (2) we can eliminate all intermixture of these codes -- a given function will return only MP error codes, or only return codes from `error-crypt.h` and/or `error-ssl.h`.

I prefer (1) -- just put the MP codes into the big `enum` in `error-crypt.h`, with their current values, unifying the namespaces with no fuss.

there are probably some other correct ways to fix it.

Reply:
(global fix made this kludge unnecessary -- reverted)